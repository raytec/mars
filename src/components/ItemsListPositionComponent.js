import React from 'react';
import {Text, View, Image} from 'react-native';

export default class ItemsListPositionComponent extends React.Component {

    constructor(props) {
        super(props);
    }

    render() {
        if (this.props.item.id === undefined) {
            return ( <View><Text>Empty item</Text></View> );
        }

        return (
            <View style={{flexDirection: 'row', flex: 1}}>
                <View>
                    <Image data-test='item-details-image'
                           style={{width: 170, height: 110, padding: 10, margin: 10}}
                           source={{uri:this.props.item.img_src}}
                    />
                </View>
                <View data-test='item-details-desc'>
                    <Text data-test="item-label" style={{fontSize: 12,padding: 0, margin: 0}}>
                        {'\n'}Vehicle name:
                    </Text>
                    <Text>{this.props.item.rover.name}</Text>
                    <Text style={{fontSize: 12}}>
                        {'\n'}Photo id:
                    </Text>
                    <Text style={{fontSize: 16}}>
                        {this.props.item.id}
                    </Text>
                </View>
            </View>
        );
    }
}

