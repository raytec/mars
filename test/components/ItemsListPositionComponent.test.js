import React from 'react';
import {shallow} from 'enzyme';
import {Text} from 'react-native';
import ItemsListPositionComponent from '../../src/components/ItemsListPositionComponent';
import {getItem} from '../ItemDataSet';

describe('Rendering item list position', function() {

    it('item list position with item', function() {
        let item = getItem(123, 'testName', 'image1');
        let component = shallow(<ItemsListPositionComponent item={item} />)
        expect(
            component
        ).toMatchSnapshot();
    });

    it('item list position with empty item', function() {
        let component = shallow(<ItemsListPositionComponent item={{}} />)
        expect(
            component.contains([
                <Text>Empty item</Text>
            ])
        ).toMatchSnapshot();
    });

});