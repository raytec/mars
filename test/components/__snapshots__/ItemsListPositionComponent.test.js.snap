// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering item list position item list position with empty item 1`] = `true`;

exports[`Rendering item list position item list position with item 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ItemsListPositionComponent
    item={
      Object {
        "id": 123,
        "img_src": "image1",
        "rover": Object {
          "name": "testName",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Component>
          <Image
            data-test="item-details-image"
            source={
              Object {
                "uri": "image1",
              }
            }
            style={
              Object {
                "height": 110,
                "margin": 10,
                "padding": 10,
                "width": 170,
              }
            }
          />
        </Component>,
        <Component
          data-test="item-details-desc"
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            data-test="item-label"
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 12,
                "margin": 0,
                "padding": 0,
              }
            }
          >
            

            Vehicle name:
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            testName
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 12,
              }
            }
          >
            

            Photo id:
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 16,
              }
            }
          >
            123
          </Text>
        </Component>,
      ],
      "style": Object {
        "flex": 1,
        "flexDirection": "row",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Image
            data-test="item-details-image"
            source={
              Object {
                "uri": "image1",
              }
            }
            style={
              Object {
                "height": 110,
                "margin": 10,
                "padding": 10,
                "width": 170,
              }
            }
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "data-test": "item-details-image",
            "source": Object {
              "uri": "image1",
            },
            "style": Object {
              "height": 110,
              "margin": 10,
              "padding": 10,
              "width": 170,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Text
              accessible={true}
              allowFontScaling={true}
              data-test="item-label"
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 12,
                  "margin": 0,
                  "padding": 0,
                }
              }
            >
              

              Vehicle name:
            </Text>,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              testName
            </Text>,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 12,
                }
              }
            >
              

              Photo id:
            </Text>,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 16,
                }
              }
            >
              123
            </Text>,
          ],
          "data-test": "item-details-desc",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": Array [
                "
",
                "Vehicle name:",
              ],
              "data-test": "item-label",
              "ellipsizeMode": "tail",
              "style": Object {
                "fontSize": 12,
                "margin": 0,
                "padding": 0,
              },
            },
            "ref": null,
            "rendered": Array [
              "
",
              "Vehicle name:",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "testName",
              "ellipsizeMode": "tail",
            },
            "ref": null,
            "rendered": "testName",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": Array [
                "
",
                "Photo id:",
              ],
              "ellipsizeMode": "tail",
              "style": Object {
                "fontSize": 12,
              },
            },
            "ref": null,
            "rendered": Array [
              "
",
              "Photo id:",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": 123,
              "ellipsizeMode": "tail",
              "style": Object {
                "fontSize": 16,
              },
            },
            "ref": null,
            "rendered": 123,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Component>
            <Image
              data-test="item-details-image"
              source={
                Object {
                  "uri": "image1",
                }
              }
              style={
                Object {
                  "height": 110,
                  "margin": 10,
                  "padding": 10,
                  "width": 170,
                }
              }
            />
          </Component>,
          <Component
            data-test="item-details-desc"
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              data-test="item-label"
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 12,
                  "margin": 0,
                  "padding": 0,
                }
              }
            >
              

              Vehicle name:
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              testName
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 12,
                }
              }
            >
              

              Photo id:
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 16,
                }
              }
            >
              123
            </Text>
          </Component>,
        ],
        "style": Object {
          "flex": 1,
          "flexDirection": "row",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Image
              data-test="item-details-image"
              source={
                Object {
                  "uri": "image1",
                }
              }
              style={
                Object {
                  "height": 110,
                  "margin": 10,
                  "padding": 10,
                  "width": 170,
                }
              }
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "data-test": "item-details-image",
              "source": Object {
                "uri": "image1",
              },
              "style": Object {
                "height": 110,
                "margin": 10,
                "padding": 10,
                "width": 170,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Text
                accessible={true}
                allowFontScaling={true}
                data-test="item-label"
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 12,
                    "margin": 0,
                    "padding": 0,
                  }
                }
              >
                

                Vehicle name:
              </Text>,
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              >
                testName
              </Text>,
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 12,
                  }
                }
              >
                

                Photo id:
              </Text>,
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 16,
                  }
                }
              >
                123
              </Text>,
            ],
            "data-test": "item-details-desc",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": Array [
                  "
",
                  "Vehicle name:",
                ],
                "data-test": "item-label",
                "ellipsizeMode": "tail",
                "style": Object {
                  "fontSize": 12,
                  "margin": 0,
                  "padding": 0,
                },
              },
              "ref": null,
              "rendered": Array [
                "
",
                "Vehicle name:",
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "testName",
                "ellipsizeMode": "tail",
              },
              "ref": null,
              "rendered": "testName",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": Array [
                  "
",
                  "Photo id:",
                ],
                "ellipsizeMode": "tail",
                "style": Object {
                  "fontSize": 12,
                },
              },
              "ref": null,
              "rendered": Array [
                "
",
                "Photo id:",
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": 123,
                "ellipsizeMode": "tail",
                "style": Object {
                  "fontSize": 16,
                },
              },
              "ref": null,
              "rendered": 123,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
